lib/portlayer/exec/base.go

```diff
@@ -72,9 +73,22 @@ func newBase(vm *vm.VirtualMachine, c *types.VirtualMachineConfigInfo, r *types.
 	// construct a working copy of the exec config
 	if c != nil && c.ExtraConfig != nil {
 		src := vmomi.OptionValueSource(c.ExtraConfig)
-		extraconfig.Decode(src, base.ExecConfig)
+		containerExecKeyValues := vmomi.OptionValueMap(c.ExtraConfig)
+
+		migratedConf, isMigration, err := migration.MigrateContainerConfig(containerExecKeyValues)
+		//TODO: We need to decide how we want to handle "partial migrations" since it is very possible the container could be broken at the end of it.
+		if isMigration && err == nil {
```

Comment:
before it, check if err != nil first. In that case, we should return err with user friendly error message, cause that might go back as docker command error message.